website/static/js/registrationUtils.js

```diff
@@ -260,10 +260,15 @@ var MetaSchema = function(params) {
     self.fulfills = params.fulfills || [];
     self.messages = params.messages || {};
 
+    var count = 0;
     $.each(self.schema.pages, function(i, page) {
         var mapped = {};
         $.each(page.questions, function(qid, question) {
-            mapped[qid] = new Question(question, qid);
+            // This is necessary now since page.questions is a list
```

Comment:
Ah, I see- the questions are nested. Ok then.

We really need some documentation of what schemas are supposed to look like; I'm a little surprised that the unique question identifier (mapping to the schema) is designated somewhere outside the schema. That seems like an invitation to bugs.